androidApp/src/main/java/com/example/tasky/android/login/screen/LoginScreen.kt

```diff
@@ -1,23 +1,164 @@
+import com.example.tasky.android.login.components.LoginTextField
+import com.example.tasky.android.theme.Black
+import com.example.tasky.android.theme.Gray
+import com.example.tasky.android.theme.LightBlue
+import com.example.tasky.android.theme.MyApplicationTheme
 import kotlinx.serialization.Serializable
 
 fun NavGraphBuilder.loginScreen() {
     composable<Login> {
-        LoginScreen()
+        LoginScreen(
+            LoginScreenState(),
+            object : LoginScreenEvents {
+                override fun onClickToSignUp() {
+                }
+
+                override fun onClickLogin() {
+                }
+
+                override fun updateLoginScreenState(newState: LoginScreenState) {
+                }
+            },
+        )
     }
 }
 
 @Serializable
 object Login
 
+data class LoginScreenState(
+    val email: String = "",
+    val isEmailValid: Boolean = false,
+    val password: String = "",
+    val showPassword: Boolean = false,
+)
+
+interface LoginScreenEvents {
+    fun onClickToSignUp()
+
+    fun onClickLogin()
+
+    fun updateLoginScreenState(newState: LoginScreenState)
+}
+
 @Composable
-internal fun LoginScreen(modifier: Modifier = Modifier) {
-    Text(Greeting().greet(), modifier)
+private fun LoginScreen(
+    state: LoginScreenState,
+    events: LoginScreenEvents,
+    modifier: Modifier = Modifier,
+) {
+    val annotatedString =
+        buildAnnotatedString {
+            append("DON'T HAVE AN ACCOUNT? ")
+            withLink(
+                LinkAnnotation.Clickable(
+                    "tag",
+                    styles =
+                        TextLinkStyles(
+                            style = SpanStyle(color = LightBlue),
+                        ),
+                    linkInteractionListener = {
+                        events.onClickToSignUp()
+                    },
+                ),
+            ) {
+                append("SIGN UP")
+            }
+        }
+    Column(
+        modifier.fillMaxSize().background(Black),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        Spacer(Modifier.height(47.dp))
+        Text("Welcome Back!", style = typography.displayMedium, lineHeight = 30.sp, color = Color.White)
+        Spacer(Modifier.height(40.dp))
+        Column(
+            Modifier
+                .fillMaxSize()
+                .background(
+                    Color.White,
+                    RoundedCornerShape(topStart = 30.dp, topEnd = 30.dp),
+                ).padding(horizontal = 16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Spacer(Modifier.height(50.dp))
+            LoginTextField(
+                state.email,
+                {
+                    events.updateLoginScreenState(state.copy(email = it))
+                },
+                "Email address",
+                state.isEmailValid,
+                null,
+                modifier = Modifier.fillMaxWidth(),
```

Comment:
I'd recommend to always use named parameters for everything. For example, I have no idea what this `null` stands for.